### PR TITLE
Arc2d line intersection

### PR DESCRIPTION
**Pull request by AxelRinghausen**
* **Please check if the PR fulfills these requirements**
- [x] The commit messages follows our guidelines (explicit, in english)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] CHANGELOG.md has been updated

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
- [x] Bug fix:
- [ ] New features:
- [ ] Documentation
- [x] Optimization

* **Other information**:

### CODE REVIEW

File: volmdlr/edges.py [hunk 1]
```diff
@@ -1591,16 +1591,55 @@ def point_distance(self, point):
...
+                                                         vector_end)
+                point_angle = volmdlr.core.clockwise_angle(vector_start,
+                                                           vector_point)
+            if point_angle <= arc_angle:
+                return True
+        return False
+
+    # def to_circle(self):
+    #     return volmdlr.wires.Circle2D(self.center, self.radius)
+
```
Comment (WirajanDASILVA) [inline volmdlr/edges.py:1620]: commented code left on purpose?

File: volmdlr/edges.py [hunk 2]
```diff
@@ -1591,16 +1591,55 @@ def point_distance(self, point):
...
+    def to_full_arc_2d(self):
+        return FullArc2D(center=self.center,
+                         start_end=self.point_at_abscissa(0),
+                         name=self.name)
 
     def line_intersections(self, line2d: Line2D):
-        circle = self.to_circle()
-        circle_intersection_points = circle.line_intersections(line2d)
+        # circle = self.to_circle()
+        # circle_intersection_points = circle.line_intersections(line2d)
```
Comment (WirajanDASILVA) [inline volmdlr/edges.py:1628]: commented code left on purpose?

File: volmdlr/edges.py [hunk 3]
```diff
@@ -2131,29 +2151,76 @@ def cut_between_two_points(self, point1, point2):
...
+            return [pt1 + t1 * vec,
+                    pt1 + t2 * vec]
         else:
-            P1 = line2d.points[0]
-            V = line2d.points[1] - line2d.points[0]
-        a = V.dot(V)
-        b = 2 * V.dot(P1 - Q)
-        c = P1.dot(P1) + Q.dot(Q) - 2 * P1.dot(Q) - self.radius ** 2
+            return []
+
```
Comment (WirajanDASILVA) [inline volmdlr/edges.py:2186]: The last else statement may be removed

**Review by WirajanDASILVA — CHANGES_REQUESTED** [review]

File: volmdlr/edges.py (hunk 1 above)
Comment (AxelRinghausen) [reply volmdlr/edges.py:1620]: Yes

File: volmdlr/edges.py (hunk 2 above)
Comment (AxelRinghausen) [reply volmdlr/edges.py:1628]: Yes

File: volmdlr/edges.py (hunk 3 above)
Comment (AxelRinghausen) [reply volmdlr/edges.py:2186]: Done in https://github.com/Dessia-tech/volmdlr/pull/405/commits/f257d57f5c14af60dcfb425b07ca0e4d5b95b504

**Review by WirajanDASILVA — APPROVED** [review]

**Pull request merged**